broadinstitute/orsp-pub
# BTRX - 722 - Replace TinyMce

## Addresses
[BTRX-722](https://broadinstitute.atlassian.net/browse/BTRX-722): Replace Tinymcs editor with a react implementation
[BTRX-502](https://broadinstitute.atlassian.net/browse/BTRX-502): Implement Comments tab on Project view. 
[BTRX-507](https://broadinstitute.atlassian.net/browse/BTRX-507): Implement Comments tab on Consent Group view. 

## Changes
- Use React library for tinyMce and remove old grails implementations.
- Make `Comments Controller` available as a resource. Replace Gorm call to get all comments to a sql query.
- Re-implement grails "comments" tab to react. On this code we introduced [React Bootstrap Table 2](https://react-bootstrap-table.github.io/react-bootstrap-table2/) which is the first step to update previous  version of this library

## Testing
- From an existing Project or Sample Data Cohort, navigate to `Comments` tab. Text area editor should be properly loaded, as the table of previous comments (if there is any) below this text area. 
- Enter any comment on the textarea and use styling from the toolkit, press Add button, the expected behaviour here is that the message appears in the table. The page shouldn't re load the entire browser window.
- If any error occurs while trying to add a new comment, an alert text should be visible indicating the user to try again later.
- Add button is disabled if the comment's textarea is empty.

---
